test/framework/src/main/java/org/elasticsearch/ingest/TestIngestDocument.java

```diff
@@ -20,14 +21,51 @@
  * Construct ingest documents for testing purposes
  */
 public class TestIngestDocument {
+    public static final long DEFAULT_VERSION = 12345L;
+    private static String VERSION = IngestDocument.Metadata.VERSION.getFieldName();
 
     /**
      * Create an IngestDocument for testing that pass an empty mutable map for ingestMetaata
      */
-    public static IngestDocument ofSourceAndMetadata(Map<String, Object> sourceAndMetadata) {
+    public static IngestDocument ofSourceAndMetadataWithNullableVersion(Map<String, Object> sourceAndMetadata) {
+        return ofSourceAndIngestWithNullableVersion(sourceAndMetadata, new HashMap<>());
+    }
+
+    /**
+     * Create an {@link IngestDocument} from the given sourceAndMetadata and ingestMetadata and a version validator that allows null
+     * _versions.  Normally null _version is not allowed, but many tests don't care about that invariant.
+     */
+    public static IngestDocument ofSourceAndIngestWithNullableVersion(
+        Map<String, Object> sourceAndMetadata,
+        Map<String, Object> ingestMetadata
+    ) {
+        Map<String, BiConsumer<String, Object>> validators = replaceValidator(VERSION, IngestSourceAndMetadata::longValidator);
+        Tuple<Map<String, Object>, Map<String, Object>> sm = IngestSourceAndMetadata.splitSourceAndMetadata(sourceAndMetadata);
+        IngestSourceAndMetadata withNullableVersion = new IngestSourceAndMetadata(sm.v1(), sm.v2(), null, validators);
+        return new IngestDocument(withNullableVersion, ingestMetadata);
+    }
+
+    /**
+     * Create an {@link IngestDocument} with {@link #DEFAULT_VERSION} as the _version metadata, if _version is not already present.
+     */
+    public static IngestDocument ofSourceAndMetadataWithDefaultVersion(Map<String, Object> sourceAndMetadata) {
```

Comment:
If this is the common case, can we shorten the method name to the original name? These are extremely long...

Reply:
`ofSourceAndMetadataWithDefaultVersion` -> `withDefaultVersion`
`ofSourceAndMetadataWithNullableVersion` -> `withNullableVersion`
`ofSourceAndIngestWithNullableVersion` -> `ofIngestWithNullableVersion`
`emptyIngestDocumentWithDefaultVersion` -> `emptyIngestDocument`